cdap-app-fabric/src/main/java/co/cask/cdap/app/preview/PreviewServer.java

```diff
@@ -0,0 +1,287 @@
+    // create cConf and hConf and set appropriate settings for preview
+    CConfiguration previewcConf = CConfiguration.copy(cConf);
+    previewcConf.set(Constants.CFG_LOCAL_DATA_DIR, previewcConf.get(Constants.CFG_LOCAL_PREVIEW_DIR));
+    previewcConf.set(Constants.Dataset.DATA_DIR, previewcConf.get(Constants.CFG_LOCAL_PREVIEW_DIR));
+    Configuration previewhConf = new Configuration(hConf);
+    File localPreviewDir = new File(cConf.get(Constants.CFG_LOCAL_PREVIEW_DIR));
+    previewcConf.set(Constants.CFG_LOCAL_DATA_DIR, localPreviewDir.getAbsolutePath());
+    previewcConf.setIfUnset(Constants.CFG_DATA_LEVELDB_DIR, "data/preview");
+    return Guice.createInjector(
+      new ConfigModule(previewcConf, previewhConf),
+      new IOModule(),
+      new AuthenticationContextModules().getMasterModule(),
+      // todo : figure if we should we share secure store modules to allow reading the secure data during preview
+      new SecurityModules().getStandaloneModules(),
+      new SecureStoreModules().getStandaloneModules(),
+      new PreviewDiscoveryRuntimeModule(discoveryService),
+      new LocationRuntimeModule().getStandaloneModules(),
+      new ConfigStoreModule().getStandaloneModule(),
+      new PreviewServerModule(),
+      new ProgramRunnerRuntimeModule().getStandaloneModules(),
+      new PreviewModules().getDataFabricModule(transactionManager),
+      new PreviewModules().getDataSetsModule(datasetFramework),
+      new DataSetServiceModules().getStandaloneModules(),
+      new MetricsClientRuntimeModule().getStandaloneModules(),
+      new LoggingModules().getStandaloneModules(),
+      new NamespaceStoreModule().getStandaloneModules(),
+      new RemoteSystemOperationsServiceModule(),
+      new AbstractModule() {
+        @Override
+        protected void configure() {
+          bind(ArtifactRepository.class).toInstance(artifactRepository);
+          bind(ArtifactStore.class).toInstance(artifactStore);
+          bind(AuthorizerInstantiator.class).toInstance(authorizerInstantiator);
+          bind(AuthorizationEnforcer.class).toInstance(authorizationEnforcer);
+          bind(PrivilegesManager.class).toInstance(privilegesManager);
+          bind(StreamAdmin.class).toInstance(streamAdmin);
+          bind(StreamConsumerFactory.class).toInstance(streamConsumerFactory);
+          bind(StreamCoordinatorClient.class).toInstance(streamCoordinatorClient);
+          // bind explore client to mock.
+          bind(ExploreClient.class).to(MockExploreClient.class);
+        }
+      });
+  }
+
+  /**
+   * starts preview services
+   */
+  @Override
+  protected void startUp() throws Exception {
+    metricsCollectionService.startAndWait();
+    datasetService.startAndWait();
+
+    // It is recommended to initialize log appender after datasetService is started,
+    // since log appender instantiates a dataset.
+    logAppenderInitializer.initialize();
```

Comment:
so metrics tables are created under the preview path, as all system datasets created during preview go there. this is handled by PreviewDatasetFramework. 

but logs are currently written to the same cdad log files as standalone. since they are using the same logback files.